### ISSUE TITLE
Authentication failure in yarn build task

### ISSUE THREAD

**Issue by paramosr** [openwisp/ansible-openwisp-wifi-login-pages#36]
Hi,
My server is Ubuntu 22.04 and I am trying to configure  _ansible-openwisp-wifi-login-pages_.

I configured the playbook.yml as in the documentation:

`- hosts: openwisp-wifi-login-pages
  become: "{{ become | default('yes') }}"
  roles:
    - openwisp.wifi_login_pages
  vars:
    wifi_login_pages_domains: ["wifi.openwisp.org"]
`

also the hosts file:

`[openwisp-wifi-login-pages]
openwisp-wifi-login-pages.mydomain.com
`
when I try to run the playbook in my server, I face the following error in the yarn build task:

```
TASK [openwisp.wifi_login_pages : yarn build] *********************************************************************************************************************************************************************
fatal: [openwisp-wifi-login-pages.mydomain.com]: FAILED! => {"msg": "Failed to set permissions on the temporary files Ansible needs to create when becoming an unprivileged user (rc: 1, err: chmod: invalid mode: 'A+user:www-data:rx:allow'\nTry 'chmod --help' for more information.\n}). For information on working around this, see https://docs.ansible.com/ansible-core/2.13/user_guide/become.html#risks-of-becoming-an-unprivileged-user"}

PLAY RECAP ********************************************************************************************************************************************************************************************************
openwisp-wifi-login-pages.mydomain.com : ok=17   changed=4    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
``` 

![image](https://github.com/openwisp/ansible-openwisp-wifi-login-pages/assets/60689893/1bf534b6-76fa-4b05-9584-a4a41610c944)

seems that the www-data user dont have permissions to do all yarn build task.

Also when I check the wifi-login-pages folder, seems that everything is okay. www-data is the owner and the group.

Do you know how to solve it?

Thanks,
Pablo

**Comment by nemesifier** [comment]
This is a common ansible issue, googling this error I found this which should help:
- https://stackoverflow.com/questions/47873671/becoming-non-root-user-in-ansible-fails
- https://www.buggycoder.com/ansible-error-failed-to-set-permissions/

It doesn't look like a problem with this role but rather a problem with the fact that ansible cannot manipulate the files it creates in the /tmp dir to execute the actions of the role.

I am closing because I think it's not a bug but feel free to reply here or supply more information.

**Comment by paramosr** [comment]
Thanks your your answer.

to fix the permissions issue, just follow the steps [you mention before](https://www.buggycoder.com/ansible-error-failed-to-set-permissions/)

Basically it is to add an _install acl_ task in our playbook.

While fixing that, I found another error related to node, which was failing me in the yarn build step.  
The error message was as follows.
`node: --openssl-legacy-provider is not allowed in NODE_OPTIONS`

To solve this, I just had to remove the version of node installed on the server (16 in my case) and install node 20.

By doing that, ansible wifi login pages is completed sucessfully.
